### ISSUE TITLE
brewtarget crashed, now I am warned there are two instances running

### ISSUE THREAD

**Issue by mikfire** [Brewtarget/brewtarget#533]
So the question is how do I clear the message? I will be incredibly disappointed if the answer is "reboot".

**Comment by xzfantom** [comment]
#527 it was introduced in this PR. If you know it's not second instance, just press ignore.

**Comment by matty0ung** [comment]
I know it's mildly annoying to have to press ignore on every subsequent run of the program after a crash (until reboot, when things will be reset) but hopefully most users most of the time will not see the program crash. :)

At the cost of some complexity, we could add logic so that you would only have to press ignore on the first run after the crash.  (I would think to use the byte of shared memory to store a "number of instances running" which we decrement on shut-down etc.)  Or we could do something Linux-specific where we more directly search for other processes running.  Or the simplest might be to add an environment variable that allows you to turn off the check for other instances.

**Comment by xzfantom** [comment]
Looks like problem mostly for *nix systems as they don't delete shared memory on crash.
Here is workaround with attach/detach: https://stackoverflow.com/questions/42549904/qsharedmemory-is-not-getting-deleted-on-application-crash
May be will work

**Comment by matty0ung** [comment]
The approach suggested by @xzfantom works on my Linux machine.  If this can remove frustrations for @mikfire, I would propose to retain the check (with these modifications) on the grounds that we don't know for certain that everything works sensibly when two instances are running (unless someone has recently one extensive testing on this?)

**Comment by mikfire** [comment]
I am running tests on Linux and Windows, the two I have easy access to. 

My testing so far consists of running two instances from different terminals -- mintty on Windows and an xterm-like on Linux. I pick a recipe and try to add a miscellaneous ingredient. I am working with a default database, and very few of them have misc. ingredients, so it is just a little faster that way.

In my five or so tests on each platform, the one with the exclusive lock can write as it wishes and the one without crashes when you try to add anything. To date, I have not seen any corruption or destruction of the database. I will do more tests, because I am now curious. 

I really do feel my original analysis was and is correct. The destruction was not caused directly by running two instances. It was caused by some really wonky logic in how we handled the temp database and a really unfortunate delete. It just so happened that running two instances was a guaranteed method for exposing the problem.

Could there be a race condition here? Maybe. 
Would I really trust my testing? Nope.

And with all that said, as long as there is a reasonable way to reduce my pain, I will shut up about it. I mean, I might snicker occasionally, but I promise I will stop whingeing :)
